dds/DCPS/DomainParticipantImpl.cpp

```diff
@@ -2187,16 +2189,16 @@ DomainParticipantImpl::handle_exception(ACE_HANDLE /*fd*/)
       = pubPtr->delete_contained_entities();
 
       if (result != DDS::RETCODE_OK) {
-        return result;
+        ret = result;
```

Comment:
Not 100% sure, maybe the next steps will result in other errors but this gives the easiest control flow